assembly/src/tokens/stream.rs

```diff
@@ -93,10 +120,49 @@ impl<'a> TokenStream<'a> {
             }
         }
     }
+
+    pub fn take_doc_comment_at(&mut self, pos: usize) -> Option<String> {
+        let comment = self.doc_comments.remove(&pos);
+        match comment {
+            Some(comment) => comment,
+            None => None,
+        }
+    }
 }
 
 impl<'a> fmt::Display for TokenStream<'a> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         write!(f, "{:?}", &self.tokens[self.pos..])
     }
 }
+
+#[derive(Debug)]
+pub struct Comment(pub String);
+
+impl Comment {
+    pub fn append_line(&mut self, line: &str) {
+        let mut prepared_line = Self::prepare_line(DOC_COMMENT_PREFIX, line);
+        prepared_line.push('\n');
+        self.0.push_str(&prepared_line);
+    }
+
+    pub fn clear(&mut self) {
+        self.0.clear();
+    }
+
+    pub fn get_comment(&self) -> Option<String> {
+        let res = self.0.trim();
+        if res.is_empty() {
+            None
+        } else {
+            Some(String::from(res))
+        }
+    }
```

Comment:
Instead of having these two methods, I'd have something like `take_content()` method which would remove the string from the comment without having to create a new string. For example, it could look like this:
```Rust
pub fn take_content(&mut self) -> Option<String> {
    self.0.take()
}
```

Assuming `Self` is `pub struct Comment(Option<String>);`